cmd/oauth.go

```
@@ -349,72 +355,85 @@ func handleOAuthCallback(r *fastglue.Request) error {
+		if upn == "" {
+			upn = email
 		}
-		if upn, ok := claims["preferred_username"].(string); ok {
-			return upn, nil
+		if email == "" {
+			return "", "", fmt.Errorf("email not found")
 		}
-		return "", fmt.Errorf("email not found")
+		return email, upn, nil
 	case string(oauth.ProviderGoogle):
 		req, _ := http.NewRequest("GET", "https://www.googleapis.com/oauth2/v2/userinfo", nil)
```

golangci-lint (2.12.2)

[error] net/http.NewRequest must not be called. use net/http.NewRequestWithContext

(noctx)